### PR TITLE
4 bug fixes

### PR DESCRIPTION
**Pull request by andrewlee94**
## Summary/Motivation:
This PR addresses 4 bugs identified recently.

1. Generic Reactions: Gibbs energy and van 't Hoff forms of calculating equilibrium constants could not be combined.
2. Generic Properties: Scaling tools raised an `Exception` due to trying to create fugacity expressions for non-vapourisable components.
3. Generic Properties: RPP5 library had misnamed variable when checking for prior parameter creation resulting in implicit replacement of parameters.
4. Unit model: `add_state_material_balances` had indexing issues due to not checking for non-vapourisables and non-condensables.

## Changes proposed in this PR:
- Reworked Gibbs energy form to not build its own `dh_rxn`
- Added new option to component declaration to indicate that a component does not have a vapour pressure and can thus be skipped when calculating frugalities. Applied this to key models.
- Fixed typo in RPP5
- Fixed indexing of Expressions in `add_state_material_balances`

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: idaes/generic_models/properties/core/reactions/equilibrium_constant.py [hunk 1]
```diff
@@ -141,10 +142,11 @@ def build_parameters(rblock, config):
...
 
-        rblock.dh_rxn_ref = Var(
-                doc="Specific molar heat of reaction at reference state",
-                units=units["energy_mole"])
-        set_param_from_config(rblock, param="dh_rxn_ref", config=config)
+        # Check that heat of reaction is constant
+        if config.heat_of_reaction is not constant_dh_rxn:
+            raise ConfigurationError(
+                "{} calculating equilibrium constants from Gibbs energy "
+                "assumes constant heat of reaction.".format(rblock.name))
```
Comment (TimBartholomew) [inline idaes/generic_models/properties/core/reactions/equilibrium_constant.py:149]: This error message reads more like a warning. I think it should be adjusted to 

"{} calculating equilibrium constants from Gibbs energy requires the user to specify a constant heat of reaction"
 or
"{} calculating equilibrium constants from Gibbs energy assumes constant heat of reaction, but the user has not specified that the heat of reaction should be constant"

**Review by TimBartholomew — COMMENTED** [review]
Minor comment on an error message.

**Comment by eslickj** [comment]
Looks good.  I'll approve and soon as @TimBartholomew's comment is addressed.

**Comment by andrewlee94** [comment]
@eslickj @TimBartholomew The error message has been updated.

**Review by eslickj — DISMISSED** [review]

**Review by TimBartholomew — APPROVED** [review]
LGTM

**Review by Andresj89 — APPROVED** [review]
LGTM

**Review by eslickj — APPROVED** [review]

**Pull request merged**